adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m

```diff
@@ -284,38 +301,33 @@ - (nullable NSError *)loadAd:(nonnull NSString *)placement
     NSError *error = nil;
     if (delegate) {
       error = GADMAdapterVungleErrorWithCodeAndDescription(
-                                                           kGADErrorMediationAdapterError, @"A banner ad type has already been "
-                                                           @"instantiated. Multiple banner ads are not "
-                                                           @"supported with Vungle iOS SDK.");
+          kGADErrorMediationAdapterError, @"A banner ad type has already been "
+                                          @"instantiated. Multiple banner ads are not "
+                                          @"supported with Vungle iOS SDK.");
     } else {
       error = GADMAdapterVungleErrorWithCodeAndDescription(
-                                                           kGADErrorMediationAdapterError, @"Can't load ad when try to add a nil delegate.");
+          kGADErrorMediationAdapterError, @"Can't load ad when try to add a nil delegate.");
     }
     return error;
   }
 
   VungleSDK *sdk = [VungleSDK sharedSDK];
   if ([self isAdCachedForPlacementID:placement withDelegate:delegate]) {
     [delegate adAvailable];
-  } else {
-    NSError *loadError;
-    GADMAdapterVungleAdType adType = [delegate adapterAdType];
-    if (adType != GADMAdapterVungleAdTypeBanner && adType != GADMAdapterVungleAdTypeShortBanner && adType != GADMAdapterVungleAdTypeLeaderboardBanner) {
-        if (![sdk loadPlacementWithID:placement error:&loadError]) {
-              if (loadError) {
-                  return loadError;
-              }
-          }
-    } else {
-        if (![sdk loadPlacementWithID:placement withSize:[self getVungleBannerAdSizeType:adType] error:&loadError]) {
-            if ((loadError) && (loadError.code != VungleSDKResetPlacementForDifferentAdSize)) {
-                return loadError;
-            }
-        }
-    }
+    return nil;
   }
 
-  return nil;
+  NSError *loadError = nil;
+  GADMAdapterVungleAdType adType = [delegate adapterAdType];
+  if (adType != GADMAdapterVungleAdTypeBanner && adType != GADMAdapterVungleAdTypeShortBanner &&
+      adType != GADMAdapterVungleAdTypeLeaderboardBanner) {
+    [sdk loadPlacementWithID:placement error:&loadError];
```

Comment:
Vungle SDK has both a BOOL return value and an error out. Will Vungle ever return NO with a nil error, or YES with a non-nil error?

Reply:
@ericleich Hi Eric, I think if `- (BOOL)loadPlacementWithID:error:` method returns NO, it will return a non-nil error, so your implementation is correct. Thanks a lot ^_^